octodns_transip/__init__.py

```diff
@@ -256,7 +256,7 @@ def _apply(self, plan):
         for change in changes:
             record = change.new
```

Comment:
This safest fix here is likely to use change.record, https://github.com/octodns/octodns/blob/53a310ec518feef14f49f47ccc3d0d3bea16c975/octodns/record/change.py#L13-L16, which will get a record regardless of the change type.

```suggestion
            record = change.record
```